core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/text/TextScanSourceOpDesc.scala

```diff
@@ -0,0 +1,97 @@
+package edu.uci.ics.texera.workflow.operators.source.scan.text
+
+import com.fasterxml.jackson.annotation.{
+  JsonIgnoreProperties,
+  JsonProperty,
+  JsonPropertyDescription
+}
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.kjetland.jackson.jsonSchema.annotations.{
+  JsonSchemaInject,
+  JsonSchemaString,
+  JsonSchemaTitle
+}
+import edu.uci.ics.amber.engine.architecture.deploysemantics.layer.OpExecConfig
+import edu.uci.ics.texera.workflow.common.metadata.annotations.HideAnnotation
+import edu.uci.ics.texera.workflow.common.tuple.schema.{
+  Attribute,
+  AttributeType,
+  OperatorSchemaInfo,
+  Schema
+}
+import edu.uci.ics.texera.workflow.operators.source.scan.ScanSourceOpDesc
+
+import java.io.{BufferedReader, FileReader}
+import scala.collection.convert.ImplicitConversions.`iterator asScala`
+
+@JsonIgnoreProperties(value = Array("limit", "offset"))
+class TextScanSourceOpDesc extends ScanSourceOpDesc {
+
+  @JsonProperty()
+  @JsonSchemaTitle("Limit")
+  @JsonPropertyDescription("max output count")
+  @JsonDeserialize(contentAs = classOf[Int])
+  @JsonSchemaInject(
+    strings = Array(
+      new JsonSchemaString(path = HideAnnotation.hideTarget, value = "outputAsSingleTuple"),
+      new JsonSchemaString(path = HideAnnotation.hideType, value = HideAnnotation.Type.equals),
+      new JsonSchemaString(path = HideAnnotation.hideExpectedValue, value = "true")
+    )
+  )
+  var limitHideable: Option[Int] = None
+
+  @JsonProperty()
+  @JsonSchemaTitle("Offset")
+  @JsonPropertyDescription("starting point of output")
+  @JsonDeserialize(contentAs = classOf[Int])
+  @JsonSchemaInject(
+    strings = Array(
+      new JsonSchemaString(path = HideAnnotation.hideTarget, value = "outputAsSingleTuple"),
+      new JsonSchemaString(path = HideAnnotation.hideType, value = HideAnnotation.Type.equals),
+      new JsonSchemaString(path = HideAnnotation.hideExpectedValue, value = "true")
+    )
+  )
+  var offsetHideable: Option[Int] = None
+
+  @JsonProperty(defaultValue = "false")
+  @JsonPropertyDescription(
+    "scan entire text file into single output tuple, ignoring any offsets and limits"
+  )
+  var outputAsSingleTuple: Boolean = false
+
+  fileTypeName = Option("Text")
+
+  @Override
+  override def operatorExecutor(operatorSchemaInfo: OperatorSchemaInfo): OpExecConfig = {
+    filePath match {
+      case Some(path) =>
+        // get offset and max line values
+        val reader = new BufferedReader(new FileReader(path))
+        val offsetValue = offsetHideable.getOrElse(0)
+        var lines = reader.lines().iterator().drop(offsetValue)
+        if (limitHideable.isDefined) lines = lines.take(limitHideable.get)
```

Comment:
please add brackets for code readability

Reply:
done, thanks!